### ISSUE TITLE
Draw - Android crashes doing local draw plus playupdates

### ISSUE THREAD

**Issue by AndyDentFree** [realm/realm-dotnet#1020]
Android Xamarin crashing on multiple devices

Seems ok when doing drawing on that device, problem occurs when drawing on the device **and** handling remote updates.

```
[REALM] jni/../src/object-store/src/impl/results_notifier.cpp:125: [realm-core-2.2.0] Assertion failed: !changes->insertions.contains(idx)
[REALM] IMPORTANT: if you see this error, please send this log to help@realm.io.
[libc] Fatal signal 6 (SIGABRT) at 0x0000421c (code=-6), thread 16955 (lm.draw.xamarin)
```

**Comment by AndyDentFree** [comment]
Results of further testing:

It is **not** related to drawing locally as I was able to crash an Android tablet with no interaction on it.

It **does** seem to be related to having two devices generating very rapid transactions at once .

I thought two devices might be a red herring just because two devices being scribbled on rapidly are able to generate more transactions than a single device. It feels like a **serialisation** issue of some kind however - I started a device from scratch with it and had several _pages_ of thousands of points for it to replay - it wasn't until I had both devices generating points that it crashed.

So far despite diligent scribbling, I have been unable to crash an iOS device whilst using two other iOS devices as input.
